### PR TITLE
fix(create-snapshot): workaround tests directory

### PR DESCRIPTION
**Pull request by pb8o**
The test infra assumes CWD is 'ROOT/tests'. create_snapshot_artifact runs from the root '.', so some functions can fail.

Workaround by changing to tests while doing the imports.

Original error:

```
[12] Command:
cd ../src/firecracker && cargo pkgid | cut -d# -f2 | cut -d: -f2
[12] stdout:

[12] stderr:
/bin/sh: 1: cd: can't cd to ../src/firecracker

Returned error code: 2
stderr:
/bin/sh: 1: cd: can't cd to ../src/firecracker
```

Signed-off-by: Pablo Barbáchano <pablob@amazon.com>

## Changes

...

## Reason

...

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [x] All commits in this PR are signed (`git commit -s`).
- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] New `unsafe` code is documented.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md

### CODE REVIEW

File: tools/create_snapshot_artifact/main.py [hunk 1]
```diff
@@ -15,6 +15,8 @@
 sys.path.append(os.path.join(os.getcwd(), "tests"))  # noqa: E402
```
Comment (roypat) [inline tools/create_snapshot_artifact/main.py:15]: Shouldn't this line already ensure that we can import relative to the tests directory?

Comment (andreitraistaru) [reply tools/create_snapshot_artifact/main.py:15]: I think this just adds a new path for the python interpreter to look for files from which to do the import but it doesn't actually change the current working directory. The method `get_firecracker_version_from_toml()` expects the working directory to be set to /firecracker/tests, it doesn't look to the paths interpreter uses

**Review by andreitraistaru — APPROVED** [review]

File: tools/create_snapshot_artifact/main.py (hunk 1 above)
Comment (andreitraistaru) [reply tools/create_snapshot_artifact/main.py:15]: I think this just adds a new path for the python interpreter to look for files from which to do the import but it doesn't actually change the current working directory. The method get_firecracker_version_from_toml() expects the working directory to be set to /firecracker/tests, it doesn't look to the paths interpreter uses

Comment (pb8o) [reply tools/create_snapshot_artifact/main.py:15]: Yes, that allows importing, but it does not move into the `tests` directory. A lot of it just assumes we are in `tests`. A better fix would be to extract the functions that are not related to testing into proper Python modules, and/or to fix the test infra to not assume the test directory. But both are pretty large in scope.

Comment (roypat) [reply tools/create_snapshot_artifact/main.py:15]: Ah, I see, slightly misunderstood what the issue was (thought it was about the imports failing)

**Review by roypat — APPROVED** [review]

**Pull request merged**